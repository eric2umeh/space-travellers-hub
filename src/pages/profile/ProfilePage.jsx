import React from 'react';
import { useSelector } from 'react-redux';
import classes from './ProfilePage.module.css';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets.rockets);
  const missions = useSelector((state) => state.missions.missions);

  const reservedMissions = missions.filter((mission) => mission.reserved !== false);

  const reservedRockets = rockets.filter((rocket) => rocket.reserved !== false);

  let missionsContents = <p>No Missions Found! Please join a new one</p>;
  if (reservedMissions.length > 0) {
    missionsContents = reservedMissions.map((mission) => (
      <li key={mission.mission_id}>{mission.mission_name}</li>
    ));
  }

  let rocketsContents = <p>No Rockets Found! Please reserved a new one!</p>;
  if (reservedRockets.length > 0) {
    rocketsContents = reservedRockets.map((rocket) => (
      <li key={rocket.rocket_id}>{rocket.rocket_name}</li>
    ));
  }

  return (
    <section className={classes.my_profile_section}>
      <div className={classes.my_missions}>
        <h3>My Missions</h3>
        <ul className={classes.profile_items}>{missionsContents}</ul>
      </div>
      <div className={classes.my_rockets}>
        <h3>My Rockets</h3>
        <ul className={classes.profile_items}>{rocketsContents}</ul>
      </div>
    </section>
  );
};

export default Profile;
