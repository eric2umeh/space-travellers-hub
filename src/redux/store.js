import { configureStore } from '@reduxjs/toolkit';
import rockets from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rockets.reducer,
  },
});

export default store;
