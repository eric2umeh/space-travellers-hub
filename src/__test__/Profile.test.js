import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import ProfilePage from '../pages/ProfilePage';
import store from '../redux/store';

it('rockets list renders properly', () => {
  const tree = renderer
    .create(<Provider store={store}><ProfilePage /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
