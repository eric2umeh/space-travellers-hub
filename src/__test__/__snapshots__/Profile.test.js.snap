// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rockets list renders properly 1`] = `
<section
  className="my_profile_section"
>
  <div
    className="my_missions"
  >
    <h3>
      My Missions
    </h3>
    <ul
      className="profile_items"
    >
      <p>
        No Missions Found! Please join a new one
      </p>
    </ul>
  </div>
  <div
    className="my_rockets"
  >
    <h3>
      My Rockets
    </h3>
    <ul
      className="profile_items"
    >
      <p>
        No Rockets Found! Please reserved a new one!
      </p>
    </ul>
  </div>
</section>
`;
